wordpress-mobile/WordPress-Android
# [Reader IA] Integrate Top Bar Filter UI to Filter functionality

Fixes #19855 
Fixes #19851

> [!Caution]
> Should not be merged before #19844

This integrates the new Filter component with the existing Filter functionality, while also updating the Filter Sheet to look a bit closer to the new specs, but some things, like the Edit Button and the style update for site items will be done in future PRs.

As a side effect, this PR also fixes #19851 since the root cause of that bug also caused the feed to refresh when selecting a Filter, which was needed for this PR.


https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/60f2c56d-6896-4e9c-8d1e-4fe4ab2ea653


-----

## To Test:

1. Install and log into Jetpack
2. Go to the Reader
3. Go to Feeds that are "filterable", such as "Subscriptions" and "Automattic" (for A12s)
4. **Verify** the appropriate filter chips appear, with the correct count inside them
5. Tap one of the filters
6. **Verify** the correct bottom sheet opens (for blogs or tags)
7. Select a filter item in the list
8. **Verify** the correct blog/tag title is shown when selected
9. Tap the filtered title (NOT the X button)
10. **Verify** the bottom sheet opens for the same type of item (blog/tag)
11. Cancel selection or select something else
12. Tap the X button in the Filter chip for the selected item
13. **Verify** clearing the filter works properly

-----

## Regression Notes

1. Potential unintended areas of impact

    - Fetching posts / Loading functionality inside the feeds

14. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual Testing

15. What automated tests I added (or what prevented me from doing so)

    - TODO

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
